src/cbc_sdk/endpoint_standard/usb_device_control.py

```diff
@@ -0,0 +1,815 @@
+
+        Args:
+            tail_end (str): String to be appended to the end of the generated URL.
+
+        Returns:
+            str: The complete URL.
+        """
+        url = self._doc_class.urlobject.format(self._cb.credentials.org_key) + tail_end
+        return url
+
+    def _count(self):
+        """
+        Returns the number of results from the run of this query.
+
+        Returns:
+            int: The number of results from the run of this query.
+        """
+        if self._count_valid:
+            return self._total_results
+
+        url = self._build_url("/_search")
+        request = self._build_request(0, -1)
+        resp = self._cb.post_object(url, body=request)
+        result = resp.json()
+
+        self._total_results = result["num_found"]
+        self._count_valid = True
+
+        return self._total_results
+
+    def _perform_query(self, from_row=0, max_rows=-1):
+        """
+        Performs the query and returns the results of the query in an iterable fashion.
+
+        Args:
+            from_row (int): The row to start the query at (default 0).
+            max_rows (int): The maximum number of rows to be returned (default -1, meaning "all").
+
+        Returns:
+            Iterable: The iterated query.
+        """
+        url = self._build_url("/_search")
+        current = from_row
+        numrows = 0
+        still_querying = True
+        while still_querying:
+            request = self._build_request(current, max_rows)
+            resp = self._cb.post_object(url, body=request)
+            result = resp.json()
+
+            self._total_results = result["num_found"]
+            self._count_valid = True
+
+            results = result.get("results", [])
+            for item in results:
+                yield self._doc_class(self._cb, item["id"], item)
+                current += 1
+                numrows += 1
+
+                if max_rows > 0 and numrows == max_rows:
+                    still_querying = False
+                    break
+
+            if current >= self._total_results:
+                break
+
+    def _run_async_query(self, context):
+        """
+        Executed in the background to run an asynchronous query.
+
+        Args:
+            context (object): Not used, always None.
+
+        Returns:
+            list: Result of the async query, which is then returned by the future.
+        """
+        url = self._build_url("/_search")
+        request = self._build_request(0, -1)
+        resp = self._cb.post_object(url, body=request)
+        result = resp.json()
+        self._total_results = result["num_found"]
+        self._count_valid = True
+        results = result.get("results", [])
+        return [self._doc_class(self._cb, item["id"], item) for item in results]
+
+    def facets(self, fieldlist, max_rows=0):
+        """
+        Return information about the facets for all known USB devices, using the defined criteria.
```

Comment:
I like the way this is implemented. We may want to consider if we want to retrospectively go back and modify other facet implementations so we have a consistent experience

Reply:
This was mostly borrowed from one of the existing facets() implementations.